src/otx/core/model/module/base.py

```diff
@@ -105,7 +108,58 @@ def configure_optimizers(self) -> dict[str, Any]:
             }
         return {"optimizer": optimizer}
 
+    def register_load_state_dict_pre_hook(self, model_classes: list[str], ckpt_classes: list[str]) -> None:
+        """Register self.model's load_state_dict_pre_hook.
+
+        Args:
+            model_classes (list[str]): Class names from training data.
+            ckpt_classes (list[str]): Class names from checkpoint state dictionary.
+        """
+        self.model.register_load_state_dict_pre_hook(model_classes, ckpt_classes)
+
+    def state_dict(self) -> dict[str, Any]:
+        """Return state dictionary of model entity with meta information.
+
+        Returns:
+            A dictionary containing datamodule state.
+
+        """
+        state_dict = super().state_dict()
+        state_dict["meta_info"] = self.meta_info
+        return state_dict
+
+    def load_state_dict(self, state_dict: dict[str, Any]) -> None:
+        """Load state dictionary from checkpoint state dictionary.
+
+        If checkpoint's meta_info and OTXLitModule's meta_info are different,
+        load_state_pre_hook for smart weight loading will be registered.
+        """
+        ckpt_meta_info = state_dict.pop("meta_info", None)
+        if ckpt_meta_info and self.meta_info and ckpt_meta_info != self.meta_info:
```

Comment:
```suggestion
        # ckpt_meta_info exists but self.meta_info does not exists
        if ckpt_meta_info and self.meta_info is None:
            msg = "`state_dict` to load has `meta_info`, but the current model has no `meta_info`. " \ 
                       "It is recommended to set proper `meta_info` for the incremental learning case."
            warn(msg)
            return super().load_state_dict(state_dict)
        # This is the case
        if ckpt_meta_info and self.meta_info and ckpt_meta_info == self.meta_info:
            ...
            return ...
        return super().load_state_dict()
```
Please add an user warning for there is `meta_info` in `state_dict` but not in `self`. In addition, there is a return object in `load_state_dict()`. Please see https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module.load_state_dict

Reply:
Done by 5bfff3084cfdc99a987f9d3b516c488219b48ca9